### ISSUE TITLE
`fflush` doesn't work since v14 (worked in v13)

### ISSUE THREAD

**Issue by ktock** [bytecodealliance/wasmtime#7437]
### Test Case

```c
#include <stdio.h>
#include <unistd.h>

int main(){
  printf(">");

  fflush(stdout);
  char buf[10];
  read(0, buf, 10);
  printf("Got %s\n", buf);
}
```

I built this using wasi-sdk 19.0.

### Steps to Reproduce

* Compile the C code above (I used wasi-sdk 19.0)
* Run it on wasmtime

### Expected Results

fflush works and buffered string is printed.

wasmtime v13 prints the string when `fflush` is called.

```console
$ wasmtime --version
wasmtime-cli 13.0.1
$ wasmtime ./a.out
>hi
Got hi

```

### Actual Results

fflush doesn't work and buffered string isn't printed.

wasmtime v14 doesn't print the string even if `fflush` is called.

```console
$ wasmtime --version
wasmtime-cli 14.0.3
$ wasmtime ./a.out
```

(nothing is printed)

### Versions and Environment

wasmtime-cli 14.0.3

Operating system: Linux  5.15.0-87-generic 

Architecture: x86_64

**Comment by alexcrichton** [comment]
Thanks for the report! This was an accidental regression from https://github.com/bytecodealliance/wasmtime/pull/6825 and this should be fixed in #7441

**Comment by pchickey** [comment]
@ktock Thanks for this bug report, we did not have test coverage for this case but I will work on adding it.
